cmd/node_agent/na/gcp_test.go

```diff
@@ -27,28 +28,93 @@ const (
 
 // mockTokenFetcher implements the mock token fetcher.
 type mockTokenFetcher struct {
+	token        string
+	errorMessage string
 }
 
 // A mock fetcher for FetchToken.
 func (fetcher *mockTokenFetcher) FetchToken() (string, error) {
-	return token, nil
+	var err error
+
+	if len(fetcher.errorMessage) > 0 {
+		err = errors.New(fetcher.errorMessage)
```

Comment:
Use fmt.errorf()? Same for all other places in this PR. So that you can remove the import "errors".

Reply:
Updated.